src/reactviews/pages/QueryResult/table/plugins/headerFilter.plugin.ts

```diff
@@ -454,18 +460,130 @@ export class HeaderFilter<T extends Slick.SlickData> {
                 filterValues: [],
                 sorted: SortProperties.NONE,
             };
+            if (this.queryResultState.state.uri) {
+                if (store.has(this.queryResultState.state.uri)) {
+                    let gridColumnMapArray = store.get(
+                        this.queryResultState.state.uri!,
+                    ) as GridColumnMap[];
+                    gridColumnMapArray = this.clearFilterValues(
+                        gridColumnMapArray,
+                        columnDef.id!,
+                    );
+                    store.set(
+                        this.queryResultState.state.uri,
+                        gridColumnMapArray,
+                    );
+                }
+            }
         } else {
             columnFilterState = {
                 columnDef: this.columnDef.id!,
                 filterValues: this.columnDef.filterValues!,
                 sorted: this.columnDef.sorted,
             };
         }
-        this.updateState(columnFilterState);
+
+        this.updateState(columnFilterState, columnDef.id!);
+    }
+
+    private updateState(newState: ColumnFilterState, columnId: string): void {
+        let newStateArray: GridColumnMap[];
+        //Check if there is any existing filter state
+        if (!this.queryResultState.state.uri) {
+            console.log("no uri set for query result state");
+            return;
+        }
+        const currentFiltersArray = store.get(
+            this.queryResultState.state.uri,
+        ) as GridColumnMap[];
+        if (currentFiltersArray) {
+            newStateArray = this.combineFilters(
+                currentFiltersArray,
+                newState,
+                columnId,
+            );
+        } else {
+            newStateArray = [{ [this.gridId]: [{ [columnId]: [newState] }] }];
+        }
```

Comment:
Could you instead use an empty array as `currentFiltersArray` if it doesn't exist in the store, and then always call `this.combineFilters()`?  It be a little more concise and easier to follow, IMO.